### ISSUE TITLE
Consider support for wasm bindgen feature

### ISSUE THREAD

**Issue by tmpfs** [time-rs/time#490]
Thanks for the library 🙏 

I was using `chrono` but migrated to `time` due to `chrono` being out of date with regards to depending upon `time:0.1.x` which in turn leads to a hard error running `cargo audit` due to a possible segfault.

I have a requirement to use `time` with `wasm-bindgen` so reverted to using `chrono` for `wasm32-unknown-unknown` as it provides the `wasmbind` feature which calls out to Javascript to initialize it's date and time.

Would you consider a PR which ports this feature from `chrono`? It would be great to see this land so I can remove `chrono` completely and of course I would be happy to put in the effort under your guidance to get this merged. Thanks 👍

**Comment by xy2iii** [comment]
Not sure I understand correctly, do you want to create a date and return it as a JS date? Something like
```rust
#[wasm_bindgen]
pub fn return_date() -> js_sys::Date {
    let date = OffsetDateTime::now_utc();
    date.into()
}
```

[Looking at the `wasmbind` feature of Chrono](https://github.com/chronotope/chrono/search?q=wasmbind), it:
- adds some code for converting from local to offset
- adds a `From<js_sys::Date>` implementation

I also saw that we had some old code to get the UTC offset back when stdweb was active:
```
    #[cfg(__time_02_cargo_web)]
    {
        use stdweb::{js, unstable::TryInto};

        let timestamp_utc = datetime.unix_timestamp();
        let low_bits = (timestamp_utc & 0xFF_FF_FF_FF) as i32;
        let high_bits = (timestamp_utc >> 32) as i32;

        let timezone_offset = js! {
            return
                new Date(((@{high_bits} << 32) + @{low_bits}) * 1000)
                    .getTimezoneOffset() * -60;
        };

        UtcOffset::seconds(timezone_offset.try_into().ok()?).ok()
    }   
```

We can maybe reuse this, but using `js-sys` instead.

**Comment by xy2iii** [comment]
#224 was the old PR, with support for stdweb. It got removed as of 8b273435 because stdweb was unmaintainted. According to this [Reddit post](https://www.reddit.com/r/rust/comments/e7trlf/question_whats_the_difference_between_stdweb_and/):

> Functionality-wise, stdweb is an all-in-one-solution roughly equivalent to wasm_bindgen + web-sys + js-sys.

**Comment by jhpratt** [comment]
Yeah, this seems reasonable. Feel free to use the previously-existing stdweb code as inspiration.

Some pre-emptive suggestions:

- place the additional dependency behind a target gate (in `Cargo.toml`, see libc for example)
- try to avoid adding syn/quote as a dependency (even transitive) if possible

**Comment by jhpratt** [comment]
Closed via #491.
